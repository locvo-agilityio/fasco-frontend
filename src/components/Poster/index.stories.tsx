import { StoryObj, Meta } from '@storybook/react';

// Components
import Poster from '.';

const meta = {
  title: 'Components/Poster',
  component: Poster,
  parameters: {
    layout: 'centered',
  },
  render: () => (
    <div className="w-[1440px]">
      <Poster />
    </div>
  ),
  tags: ['autodocs'],
} satisfies Meta<typeof Poster>;

export default meta;

type Story = StoryObj<typeof Poster>;

export const Default: Story = {};
