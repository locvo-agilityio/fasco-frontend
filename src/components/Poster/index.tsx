'use client';

import Image from 'next/image';
import Link from 'next/link';

// Import
import PersonRightImage from '../../../public/assets/image-person-right.webp';
import PersonLeftImage from '../../../public/assets/image-person-left.webp';
import PeopleTopImage from '../../../public/assets/image-people-top.webp';
import PeopleBottomImage from '../../../public/assets/image-people-bot.webp';

// Component
import { Button } from '@/components/common';

// Constants
import { ROUTES } from '@/constants';

const Poster = () => (
  <div className="w-full h-full max-h-[756px] flex flex-col md:flex-row gap-9">
    <div className="max-w-[392px] w-full h-full bg-gray-300 rounded-regular">
      <div className="hidden relative md:block w-full min-h-[756px]">
        <Image
          alt="Fashion model seated, showcasing a stylish outfit"
          src={PersonLeftImage}
          quality={100}
          fill
          sizes="50vw"
          style={{ objectFit: 'contain' }}
          priority
        />
      </div>
    </div>

    <div className="w-full h-[756px] flex flex-col items-center justify-between gap-2">
      <div className="max-h-[150px] w-full h-full bg-gray-300 rounded-regular">
        <div className="relative w-full min-h-[150px]">
          <Image
            alt="Fashion Quartet"
            src={PeopleTopImage}
            quality={100}
            fill
            sizes="50vw"
            style={{ objectFit: 'cover' }}
            priority
          />
        </div>
      </div>
      <div className="flex flex-col items-center">
        <p className="md:text-5xl text-[80px] leading-[91px] font-poppins font-medium text-primary-200">
          ULTIMATE
        </p>
        <p
          className="md:text-6xl text-9xl leading-[187px] font-poppins font-medium text-transparent"
          style={{ WebkitTextStroke: '2px #484848' }}
        >
          SALE
        </p>

        <p className="text-md font-poppins text-primary-200">NEW COLLECTION</p>

        <Button
          as={Link}
          href={ROUTES.SHOP}
          size="md"
          radius="lg"
          className="w-52 h-14"
        >
          SHOP NOW
        </Button>
      </div>
      <div className="max-h-[150px] w-full h-full">
        <div className="relative w-full min-h-[150px]">
          <Image
            alt="Sparkling Duo"
            src={PeopleBottomImage}
            quality={100}
            fill
            sizes="50vw"
            style={{ objectFit: 'cover' }}
            priority
            className="rounded-regular"
          />
        </div>
      </div>
    </div>

    <div className="max-w-[392px] w-full h-full bg-gray-300 rounded-regular">
      <div className="hidden relative md:block w-full min-h-[756px]">
        <Image
          alt="Fashion model seated, showcasing a stylish outfit"
          src={PersonRightImage}
          quality={100}
          fill
          sizes="50vw"
          style={{ objectFit: 'contain' }}
          priority
        />
      </div>
    </div>
  </div>
);

export default Poster;
