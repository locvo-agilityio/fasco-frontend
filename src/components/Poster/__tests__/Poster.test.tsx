import { render } from '@testing-library/react';

// Components
import Poster from '..';

describe('Poster', () => {
  it('should render correctly', () => {
    const { container } = render(<Poster />);

    expect(container).toMatchSnapshot();
  });
});
