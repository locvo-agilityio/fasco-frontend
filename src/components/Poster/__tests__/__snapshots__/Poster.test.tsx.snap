// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poster should render correctly 1`] = `
<div>
  <div
    class="w-full h-full max-h-[756px] flex flex-col md:flex-row gap-9"
  >
    <div
      class="max-w-[392px] w-full h-full bg-gray-300 rounded-regular"
    >
      <div
        class="hidden relative md:block w-full min-h-[756px]"
      >
        <img
          alt="Fashion model seated, showcasing a stylish outfit"
          data-nimg="fill"
          decoding="async"
          sizes="50vw"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=100"
          srcset="/_next/image?url=%2Fimg.jpg&w=384&q=100 384w, /_next/image?url=%2Fimg.jpg&w=640&q=100 640w, /_next/image?url=%2Fimg.jpg&w=750&q=100 750w, /_next/image?url=%2Fimg.jpg&w=828&q=100 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=100 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=100 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=100 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=100 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=100 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: contain; color: transparent;"
        />
      </div>
    </div>
    <div
      class="w-full h-[756px] flex flex-col items-center justify-between gap-2"
    >
      <div
        class="max-h-[150px] w-full h-full bg-gray-300 rounded-regular"
      >
        <div
          class="relative w-full min-h-[150px]"
        >
          <img
            alt="Fashion Quartet"
            data-nimg="fill"
            decoding="async"
            sizes="50vw"
            src="/_next/image?url=%2Fimg.jpg&w=3840&q=100"
            srcset="/_next/image?url=%2Fimg.jpg&w=384&q=100 384w, /_next/image?url=%2Fimg.jpg&w=640&q=100 640w, /_next/image?url=%2Fimg.jpg&w=750&q=100 750w, /_next/image?url=%2Fimg.jpg&w=828&q=100 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=100 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=100 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=100 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=100 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=100 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
          />
        </div>
      </div>
      <div
        class="flex flex-col items-center"
      >
        <p
          class="md:text-5xl text-[80px] leading-[91px] font-poppins font-medium text-primary-200"
        >
          ULTIMATE
        </p>
        <p
          class="md:text-6xl text-9xl leading-[187px] font-poppins font-medium text-transparent"
        >
          SALE
        </p>
        <p
          class="text-md font-poppins text-primary-200"
        >
          NEW COLLECTION
        </p>
        <a
          class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 gap-2 rounded-large [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none border-default data-[hover=true]:opacity-hover border-1 bg-primary-300 text-foreground-100 disabled:opacity-50 py-5 text-base rounded-regular data-[pressed=true]:scale-[1] font-poppins w-52 h-14"
          href="/shop"
          role="button"
          tabindex="0"
        >
          SHOP NOW
        </a>
      </div>
      <div
        class="max-h-[150px] w-full h-full"
      >
        <div
          class="relative w-full min-h-[150px]"
        >
          <img
            alt="Sparkling Duo"
            class="rounded-regular"
            data-nimg="fill"
            decoding="async"
            sizes="50vw"
            src="/_next/image?url=%2Fimg.jpg&w=3840&q=100"
            srcset="/_next/image?url=%2Fimg.jpg&w=384&q=100 384w, /_next/image?url=%2Fimg.jpg&w=640&q=100 640w, /_next/image?url=%2Fimg.jpg&w=750&q=100 750w, /_next/image?url=%2Fimg.jpg&w=828&q=100 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=100 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=100 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=100 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=100 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=100 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
          />
        </div>
      </div>
    </div>
    <div
      class="max-w-[392px] w-full h-full bg-gray-300 rounded-regular"
    >
      <div
        class="hidden relative md:block w-full min-h-[756px]"
      >
        <img
          alt="Fashion model seated, showcasing a stylish outfit"
          data-nimg="fill"
          decoding="async"
          sizes="50vw"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=100"
          srcset="/_next/image?url=%2Fimg.jpg&w=384&q=100 384w, /_next/image?url=%2Fimg.jpg&w=640&q=100 640w, /_next/image?url=%2Fimg.jpg&w=750&q=100 750w, /_next/image?url=%2Fimg.jpg&w=828&q=100 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=100 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=100 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=100 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=100 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=100 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: contain; color: transparent;"
        />
      </div>
    </div>
  </div>
</div>
`;
